nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SetSNMP.java

```diff
@@ -16,209 +16,127 @@
-     */
+
     @Override
-    protected void onTriggerSnmp(ProcessContext context, ProcessSession processSession) throws ProcessException {
+    public void onTrigger(ProcessContext context, ProcessSession processSession) {
         FlowFile flowFile = processSession.get();
         if (flowFile != null) {
-            // Create the PDU object
-            PDU pdu = null;
-            if(this.snmpTarget.getVersion() == SnmpConstants.version3) {
-                pdu = new ScopedPDU();
-            } else {
-                pdu = new PDU();
-            }
-            if(this.addVariables(pdu, flowFile.getAttributes())) {
+            PDU pdu = createPdu();
+            if (SNMPUtils.addVariables(pdu, flowFile.getAttributes(), getLogger())) {
                 pdu.setType(PDU.SET);
-                try {
-                    ResponseEvent response = this.targetResource.set(pdu);
-                    if(response.getResponse() == null) {
-                        processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                        this.getLogger().error("Set request timed out or parameters are incorrect.");
-                        context.yield();
-                    } else if(response.getResponse().getErrorStatus() == PDU.noError) {
-                        flowFile = SNMPUtils.updateFlowFileAttributesWithPduProperties(pdu, flowFile, processSession);
-                        processSession.transfer(flowFile, REL_SUCCESS);
-                        processSession.getProvenanceReporter().send(flowFile, this.snmpTarget.getAddress().toString());
-                    } else {
-                        final String error = response.getResponse().getErrorStatusText();
-                        flowFile = SNMPUtils.addAttribute(SNMPUtils.SNMP_PROP_PREFIX + "error", error, flowFile, processSession);
-                        processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                        this.getLogger().error("Failed while executing SNMP Set [{}] via " + this.targetResource + ". Error = {}", new Object[]{response.getRequest().getVariableBindings(), error});
-                    }
-                } catch (IOException e) {
-                    processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                    this.getLogger().error("Failed while executing SNMP Set via " + this.targetResource, e);
-                    context.yield();
-                }
+                processPdu(context, processSession, flowFile, pdu);
             } else {
                 processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                this.getLogger().warn("No attributes found in the FlowFile to perform SNMP Set");
+                getLogger().warn("No attributes found in the FlowFile to perform SNMP Set");
```

Comment:
It would be helpful to log the FlowFile object or identifier for troubleshooting.
```suggestion
                getLogger().warn("No attributes found in {} to perform SNMP Set", flowFile);
```